src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -22,12 +22,30 @@
 
 /**
  * Compiler pass to register tagged services for an event dispatcher.
+ *
+ * @psalm-type BeforeAfterDefinition = string|array{0: string, 1: string}
+ * @psalm-type ListenerDefinition = array{serviceId: string, event: string, method: string, before?: BeforeAfterDefinition, after?: BeforeAfterDefinition, priority?: int, dispatchers: list<string>}
+ * @psalm-type AllListenersDefinition = array<string, list<ListenerDefinition>>
  */
 class RegisterListenersPass implements CompilerPassInterface
 {
     private array $hotPathEvents = [];
     private array $noPreloadEvents = [];
 
+    /**
+     * $allListenersMap['event_name']['listener_service_id']['method'] => $listenerDefinition.
+     *
+     * @var array<string, array<string, array<string, ListenerDefinition>>>
```

Comment:
using a psalm type alias in `@var` rather than `@psalm-var` is a very bad idea. another tool that does not understand the psalm-specific annotation will read this type as using a ListenerDefinition class, being totally wrong.